src/main/kotlin/io/snyk/plugin/ui/toolwindow/SnykToolWindowPanel.kt

```diff
@@ -425,22 +424,22 @@ class SnykToolWindowPanel(val project: Project) : JPanel(), Disposable {
                     // todo: open package manager file, if any and  was not opened yet
                 }
                 is SuggestionTreeNode -> {
-                    val psiFile = (node.parent as? SnykCodeFileTreeNode)?.userObject as? PsiFile
+                    val snykCodeFile = (node.parent as? SnykCodeFileTreeNode)?.userObject as? SnykCodeFile
                         ?: throw IllegalArgumentException(node.toString())
                     val (suggestion, index) = node.userObject as Pair<SuggestionForFile, Int>
 
                     val scrollPane = wrapWithScrollPane(
-                        SuggestionDescriptionPanel(psiFile, suggestion, index)
+                        SuggestionDescriptionPanel(snykCodeFile, suggestion, index)
                     )
                     descriptionPanel.add(scrollPane, BorderLayout.CENTER)
 
                     val textRange = suggestion.ranges[index]
                         ?: throw IllegalArgumentException(suggestion.ranges.toString())
-                    if (!smartReloadMode && psiFile.virtualFile.isValid) {
-                        if (textRange.start >= 0 && textRange.end < psiFile.textLength) {
-                            navigateToSource(psiFile.virtualFile, textRange.start, textRange.end)
+                    if (!smartReloadMode && snykCodeFile.virtualFile.isValid) {
+                        if (textRange.start >= 0 && textRange.end < snykCodeFile.virtualFile.length) {
```

Comment:
we can't compare position in text to `File length in bytes`. We need to get text length here.